[common] Archive.org broken ($15 bounty - backed)

[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=28378995)](https://www.bountysource.com/issues/28378995-common-archive-org-broken?utm_source=28378995&utm_medium=shield&utm_campaign=ISSUE_BADGE)

This should be as simple as taking a look at changes in the JSON output of the Archive.org API and updating our respective objects to match whatever they have changed.

If it's any more complex it could be only because they've changed the REST URL structures and then it's a matter of changing URLs.

```
FrostWire version 6.1.7 build 168
Nov 18, 2015 2:45:50 PM com.frostwire.search.PagedWebSearchPerformer 
INFO: Error searching page: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 1 column 1011 path $.response.docs[0].subject
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 1 column 1011 path $.response.docs[0].subject
at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(Unknown Source)
at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(Unknown Source)
at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(Unknown Source)
at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(Unknown Source)
at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(Unknown Source)
at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(Unknown Source)
at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(Unknown Source)
at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(Unknown Source)
at com.google.gson.Gson.fromJson(Unknown Source)
at com.google.gson.Gson.fromJson(Unknown Source)
at com.google.gson.Gson.fromJson(Unknown Source)
at com.google.gson.Gson.fromJson(Unknown Source)
at com.frostwire.util.JsonUtils.toObject(Unknown Source)
at com.frostwire.search.archiveorg.ArchiveorgSearchPerformer.searchPage(Unknown Source)
at com.frostwire.search.PagedWebSearchPerformer.searchPage(Unknown Source)
at com.frostwire.search.PagedWebSearchPerformer.perform(Unknown Source)
at com.frostwire.search.SearchManagerImpl$PerformTask.run(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 1 column 1011 path $.response.docs[0].subject
at com.google.gson.stream.JsonReader.beginArray(Unknown Source)
at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(Unknown Source)
at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(Unknown Source)
at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(Unknown Source)
... 20 more
```

Perhaps @votaguz or @X-NicON  might want to take a look?

Starting point to look is `at com.frostwire.search.archiveorg.ArchiveorgSearchPerformer.searchPage` and its related classes.

--- a/common/src/main/java/com/frostwire/search/archiveorg/ArchiveorgItem.java
+++ b/common/src/main/java/com/frostwire/search/archiveorg/ArchiveorgItem.java
@@ -39,7 +39,6 @@ public class ArchiveorgItem {
     public int num_reviews;
     public float avg_rating;
     public String identifier;
-    public List<String> subject;
     public List<String> format;
     public List<String> collection;
 }

--- a/common/src/main/java/com/frostwire/search/archiveorg/ArchiveorgSearchPerformer.java
+++ b/common/src/main/java/com/frostwire/search/archiveorg/ArchiveorgSearchPerformer.java
@@ -48,7 +48,7 @@ public class ArchiveorgSearchPerformer extends CrawlPagedWebSearchPerformer<Arch
                 + getDomainName()
                 + "/advancedsearch.php?q="
                 + encodedKeywords
-                + "&fl[]=avg_rating&fl[]=call_number&fl[]=collection&fl[]=contributor&fl[]=coverage&fl[]=creator&fl[]=date&fl[]=description&fl[]=downloads&fl[]=foldoutcount&fl[]=format&fl[]=headerImage&fl[]=identifier&fl[]=imagecount&fl[]=language&fl[]=licenseurl&fl[]=mediatype&fl[]=month&fl[]=num_reviews&fl[]=oai_updatedate&fl[]=publicdate&fl[]=publisher&fl[]=rights&fl[]=scanningcentre&fl[]=source&fl[]=subject&fl[]=title&fl[]=type&fl[]=volume&fl[]=week&fl[]=year&rows=50&page=1&indent=yes&output=json";
+                + "&fl[]=avg_rating&fl[]=call_number&fl[]=collection&fl[]=contributor&fl[]=coverage&fl[]=creator&fl[]=date&fl[]=description&fl[]=downloads&fl[]=foldoutcount&fl[]=format&fl[]=headerImage&fl[]=identifier&fl[]=imagecount&fl[]=language&fl[]=licenseurl&fl[]=mediatype&fl[]=month&fl[]=num_reviews&fl[]=oai_updatedate&fl[]=publicdate&fl[]=publisher&fl[]=rights&fl[]=scanningcentre&fl[]=source&fl[]=title&fl[]=type&fl[]=volume&fl[]=week&fl[]=year&rows=50&page=1&indent=yes&output=json";
         //sort[]=downloads+desc&sort[]=createdate+desc
         //sort[]=avg_rating+desc&
     }

